spanner/client_test.go

```diff
@@ -1420,6 +1421,66 @@ func TestClient_ReadWriteTransactionCommitAlreadyExists(t *testing.T) {
 	}
 }
 
+func TestClient_ReadWriteTransactionConcurrentQueries(t *testing.T) {
+	t.Parallel()
+	server, client, teardown := setupMockedTestServer(t)
+	defer teardown()
+	var (
+		ctx                 = context.Background()
+		wg                  = sync.WaitGroup{}
+		firstTransactionID  transactionID
+		secondTransactionID transactionID
+	)
+	_, err := client.ReadWriteTransaction(ctx, func(ctx context.Context, tx *ReadWriteTransaction) error {
+		query := func(id *transactionID) {
+			defer func() {
+				if tx.tx != nil {
+					*id = tx.tx
+				}
+				wg.Done()
+			}()
+			iter := tx.Query(ctx, NewStatement(SelectSingerIDAlbumIDAlbumTitleFromAlbums))
+			defer iter.Stop()
+			rowCount := int64(0)
+			for {
+				row, err := iter.Next()
+				if err == iterator.Done {
+					break
+				}
+				if err != nil {
+					return
+				}
+				var singerID, albumID int64
+				var albumTitle string
+				if err := row.Columns(&singerID, &albumID, &albumTitle); err != nil {
+					return
+				}
+				rowCount++
+			}
+			return
+		}
+		wg.Add(2)
+		go query(&firstTransactionID)
+		go query(&secondTransactionID)
+		wg.Wait()
+		return nil
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	if firstTransactionID == nil || secondTransactionID == nil || string(firstTransactionID) != string(secondTransactionID) {
```

Comment:
>Verify that everything works as expected if a read/write transaction executes two queries in parallel as the first statement. Only one should include a Begin, and the other should block until the first has returned a transaction.

@olavloite is it sufficient to test this case?

Reply:
I would suggest that we also add a verification to the requests that we are seeing on the mock server, so verify that the first `ExecuteSqlRequest` on the server includes the `BeginTransaction` option, and that the second one includes a `TransactionOptions.Id`.

Reply:
Done